zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -705,12 +712,28 @@ boolean isRunning() {
     }
 
     void closeSocket() {
-        try {
-            if (sock != null) {
+        if (closeSocketAsync) {
+            new Thread(new Runnable() {
```

Comment:
What about giving a meaningful name to the thread and making it a daemon thread?

Reply:
done!